main.c

```diff
@@ -149,6 +164,14 @@ static void create_layer_surface(struct swaylock_surface *surface) {
 	zwlr_layer_surface_v1_add_listener(surface->layer_surface,
 			&layer_surface_listener, surface);
 
+	wl_surface_commit(surface->surface);
+
+	wl_callback_add_listener(
```

Comment:
Could we add a comment explaining why we sync here? Something like: "We need to wait for the xdg_output.name event to be received before we can decide which image to display."